service/controller/resource/clusterconfigmap/desired.go

```diff
@@ -58,11 +62,46 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) ([]*cor
 		}
 	}
 
+	var irsa bool
+	var accountID string
+	{
+		if r.provider == "aws" {
+
+			awsCluster := &v1alpha3.AWSCluster{}
+			err := r.ctrlClient.Get(ctx, types.NamespacedName{Name: cr.Name, Namespace: cr.Namespace}, awsCluster)
+			if err != nil {
+				return nil, microerror.Mask(err)
+			}
+
+			if _, ok := awsCluster.Annotations[k8smetadata.AWSIRSA]; ok {
+				irsa = true
+			}
+
+			secret, err := r.k8sClient.CoreV1().Secrets(awsCluster.Spec.Provider.CredentialSecret.Namespace).Get(ctx, awsCluster.Spec.Provider.CredentialSecret.Name, metav1.GetOptions{})
+			if apierrors.IsNotFound(err) {
+				r.logger.Debugf(ctx, "secret '%s/%s' not found cannot set accountID", cr.Namespace, key.APISecretName(&cr))
+			} else if err != nil {
+				return nil, microerror.Mask(err)
+			}
+			arn := string(secret.Data["aws.awsoperator.arn"])
+
+			re := regexp.MustCompile(`[-]?\d[\d,]*[\.]?[\d{2}]*`)
+			accountID = re.FindAllString(arn, 1)[0]
```

Comment:
is there a way this can panic or are we always sure the ARN will be a valid string?

Reply:
I can check if the accountID is empty so it returns with an error in case it's empty but it shouldn't do any harm or even panic. Even if the value is empty it will still work, apps will use an empty string for the accountID

Reply:
You sure?

```
re.FindAllString(arn, 1)[0]
```

`FindAllString` returns a slices, and if that's empty `[0]` should panic with `index out of range`

Reply:
Good point 👍🏻, I added a check if the arn is empty and return if this happens to prevent panicking